File: android/src/main/kotlin/com/unitedtraders/mobile/zendesk_messaging/UrlToHandleInAppStreamHandler.kt
package com.unitedtraders.mobile.zendesk_messaging

import io.flutter.plugin.common.EventChannel

class UrlToHandleInAppStreamHandler: EventChannel.StreamHandler {
    private var eventsSink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventsSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventsSink = null
    }

    fun onUrlToHandleInAppEvent(urlToHandle: String) {
        eventsSink?.success(urlToHandle)
    }
}

File: ios/Classes/UrlToHandleInAppStreamHandler.swift
//
//  UrlToHandleInAppStreamHandler.swift
//  zendesk_messaging
//
//  Created by Aleksey Shepelev on 05.07.2022.
//

import Foundation

class UrlToHandleInAppStreamHandler: NSObject, FlutterStreamHandler {
    var eventSink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }
    
    func handleUrlToHandleInAppEvent(_ urlToHandle: String) {
        eventSink?(urlToHandle)
    }
}
